src/DetailView/DetailView.js

```diff
@@ -1,53 +1,73 @@
+import { keyboard } from "@testing-library/user-event/dist/keyboard"
 import React, { Component } from "react"
 import "./DetailView.css"
+import backButton from "../assets/back-button.png"
 
 class DetailView extends Component {
   constructor(props) {
     super(props)
     this.movieId = this.props.movieId
     this.closeMovie = this.props.closeMovie
-    this.state = {
-    }
+    this.state = {}
   }
 
   componentDidMount() {
     fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${this.movieId}`)
       .then(response => response.json())
       .then(movieObject => {
         const data = movieObject.movie
+        const releaseDateObject = new Date(Date.parse(data.release_date))
+        const dateData = `${releaseDateObject.toLocaleString("default", { month: "long" })} ${releaseDateObject.getDate()}, ${releaseDateObject.getFullYear()}`
+        const dateYear = releaseDateObject.getFullYear()
 
         this.setState({
           averageRating: data.average_rating,
           backdropPath: data.backdrop_path,
-          budget: data.budget,
-          genres: data.genres,
+          budget: data.budget.toLocaleString(),
+          genres: data.genres.join(' '),
           id: data.id,
           overview: data.overview,
           posterPath: data.poster_path,
-          releaseDate: data.release_date,
-          revenue: data.revenue,
+          releaseDate: releaseDateObject,
+          dateData: dateData,
+          dateYear: dateYear,
+          revenue: data.revenue.toLocaleString(),
           runtime: data.runtime,
-          tagline: data.tagline,
           title: data.title
         })
       })
+
+    fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${this.movieId}/videos`)
+      .then(response => response.json())
+      .then(videoObject => {
+        const data = videoObject.videos[0]
+        const isEmbedAvailable = data === [] ? false : true
+        if (isEmbedAvailable) {
+          this.setState({
+            videoUrl: data.key
+          })
```

Comment:
```suggestion
      .then({ videos } => {
          if (videos.length) {
            this.setState({
                videoUrl: data.key
           })
```

I think this would also work and is a little easier to skim over.

Reply:
That looks much cleaner. I'm still getting used to object deconstructing in my workflow, anywhere you think we could incorporate it I'd love to just to build up some muscle memory there.

Reply:
Updated it to reflect this logic in the latest PR.